File: cs/lc189.cs
public class Solution {
    public void Rotate(int[] nums, int k) {
        int n = nums.Length;
        k = k % n;
        int[] ans = new int[n];
        for(int i = 0;i<n-k; i++)
        {
            ans[i+k] = nums[i];
        }
        int idx = 0;
        for(int i = n-k;i<n; i++)
        {
            ans[idx++] = nums[i];
        }
        for(int i=0;i<n;i++)
        {
            nums[i] = ans[i];
        }
    }
}

File: python/lc189.py
class Solution:
    def rotate(self, nums: List[int], k: int) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        n = len(nums)
        k = k % n
        ans = nums[n-k:] + nums[0:n-k]
        for i in range(n):
            nums[i] = ans[i]
